pkg/api/controller.go

```diff
@@ -166,12 +166,23 @@ func (c *Controller) Run(reloadCtx context.Context) error {
 	c.Server = server
 
 	// Create the listener
-	listener, err := net.Listen("tcp", addr)
-	if err != nil {
-		return err
-	}
-
-	if c.Config.HTTP.Port == "0" || c.Config.HTTP.Port == "" {
+	var listener net.Listener
+	var err error
+
+	if strings.HasPrefix(c.Config.HTTP.Address, "unix:") {
```

Comment:
There should be a test to cover this type of configuration.

Reply:
Wait, why not just listen on localhost?

Reply:
> Wait, why not just listen on localhost?

Any process on the system can reach a 'localhost' endpoint while unix sockets are protected by filesystem permissions. I agree that this isn't a *huge* win, as I could protect access to the zot via httpasswd auth and a randomly generated username/key, but that comes for free with a unix socket.

I'll accept that unix socket support isn't needed at the moment, but rejecting the use of unix sockets due to the availability of 'localhost' isn't an idea that would stand on its own very long. I really did this as a proof of concept.

Reply:
And in addition to protecting unix socket with file perms, this also opens the door for potential zot configs saying "only uid 100000-165536 can connect", using SO_PEERCRED.